### DISCUSSION TITLE
How can I add a view action in resource A to view the relation model details form from resource B?

### DISCUSSION THREAD

**Discussion by Saad5400** [filamentphp/filament#13889] [Help]
### Package

Actions

### Package Version

v3.2

### How can we help you?

As the title, I have a pretty long form for one resource, that I want to reuse (in the view action) for another resource which is related to that long model.

**Comment by OctavioCaba** [comment]
I have the same doubt, it seems community is not very active these days...

**Reply by Saad5400** [reply]
I'v ended up using a custom Action and linking (url) to the other model from the other resource.

**Comment by e2d2-dev** [comment]
Hi,

maybe just wrap your form into a function and call it.

```
public function myFormSchema(): array 
{
    return [
        // your components
    ];
}
```

**Reply by Saad5400** [reply]
So if I have this function in resource A which uses the model A.

How can I add a view action to use this function, in resource B which uses model B (and has a relationship to A)
